File: typescript/src/LicenseLynx.ts
import * as fs from 'fs';

function map(licenseName: string) {
    try {
        const mergedData = JSON.parse(fs.readFileSync('src/resources/merged_data.json', 'utf8'));
        return mergedData[licenseName] || { error: 'License not found' };
    } catch (error) {
        console.error('Error reading data:', error);
        return { error: 'Failed to read data' };
    }
}

export { map };

File: python/src/license_lynx.py
import json


class LicenseLynx:

    @staticmethod
    def map(license_name):
        try:
            with open('resources/merged_data.json', 'r') as file:
                merged_data = json.load(file)
                return merged_data.get(license_name, {"error": "License not found"})
        except FileNotFoundError:
            return {"error": "License not found"}
